### PR TITLE
Catalogs

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-2420
| Versions      | master (4.2)

Note: catalog statuses not mentioned, as they are not functional at the moment out of the box.

User guide PR: https://github.com/ezsystems/user-documentation/pull/138

To do:

- [x] Verify filter list
- [x] Add missing screenshots

### CODE REVIEW

**Review by mikadamczyk — APPROVED** [review]

File: docs/guide/catalog/catalog.md [hunk 1]
```diff
@@ -7,6 +7,23 @@ description: Product catalog offers PIM functionalities, with product, product t
...
+Use them, for example, to build special discount catalogs, campaigns, or tenders.
+
+When creating a catalog, you can filter the products by:
+
+- price
+- product attributes
+- product type
+- product code
+- availability
+- product category
```
Comment (adamwojs) [inline docs/guide/catalog/catalog.md:23]: "created at" filter is missing

File: docs/guide/catalog/catalog.md [hunk 2]
```diff
@@ -7,6 +7,23 @@ description: Product catalog offers PIM functionalities, with product, product t
...
+
+![]()
```
Comment (adamwojs) [inline docs/guide/catalog/catalog.md:25]: ???

File: code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Controller;
```
Comment (adamwojs) [inline code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php:3]:
```suggestion
declare(strict_types=1);

namespace App\Controller;
```

File: code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    public function listPromoProductsAction()
+    {
+        $catalog = $this->catalogService->getCatalogByIdentifier('desk_promo');
+        $query = new ProductQuery($catalog->getQuery());
+
+        $result = $this->productService->findProducts($query);
+
+        return $this->render('full/promo.html.twig', [
```
Comment (adamwojs) [inline code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php:29]: Please consider using ibexadesign here

File: code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        $this->productService = $productService;
+        $this->catalogService = $catalogService;
+    }
+
+    public function listPromoProductsAction()
+    {
+        $catalog = $this->catalogService->getCatalogByIdentifier('desk_promo');
+        $query = new ProductQuery($catalog->getQuery());
+
+        $result = $this->productService->findProducts($query);
```
Comment (adamwojs) [inline code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php:27]: Maybe we can add pagination/sorting to the example?

File: docs/guide/content_rendering/embed_and_list_content/list_products_in_catalog.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+To list products from a specific [catalog](../../catalog/catalog.md#catalogs), you can either use the Content query Field,
+or a custom controller.
+
+## List products with Content query Field
+
+To list products from a catalog in a [Content query Field](../../../api/field_types_reference/contentqueryfield.md),
+add this Field to a Content Type definition.
+
+Select the ["Catalog" Query type](../queries_and_controllers/built-in_query_types.md#catalog) and "Product" as returned type.
+In parameters, provide the identifier of the catalog.
```
Comment (adamwojs) [inline docs/guide/content_rendering/embed_and_list_content/list_products_in_catalog.md:16]:
```suggestion
Select the ["Catalog" Query type](../queries_and_controllers/built-in_query_types.md#catalog).
In parameters, provide the identifier of the catalog.
```

Currently there is not valid option here for 4.x as new PC works on multiple content types. Mentioned "Product" refer to deprecated 3.3 product catalog.

**Review by adamwojs — APPROVED** [review]

File: docs/guide/catalog/catalog.md (hunk 2 above)
Comment (DominikaK) [reply docs/guide/catalog/catalog.md:25]: This is an image placeholder, to be added after last ui fixes

File: code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php (hunk 5 above)
Comment (DominikaK) [reply code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php:27]: Given that this is for illustrative purposes, I prefer to keep the examples as simple as possible, even if they are not actual best practices

File: docs/guide/content_rendering/embed_and_list_content/list_products_in_catalog.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+You can also customize the rendering of products from a selected catalog with a custom controller and render it in a custom route.
+
+In this example, create a custom controller for a promo catalog in `src/Controllers/PromoController.php`:
+
+``` php hl_lines="26 27 29"
+[[= include_file('code_samples/front/shop/list_products_in_catalog/src/Controller/PromoController.php') =]]
+```
+
+The controller uses `CatalogService` to get the selected catalog by its identifier, `desk_promo` (line 24).
```
Comment (juskora) [inline docs/guide/content_rendering/embed_and_list_content/list_products_in_catalog.md:35]: After the doc build, it shows lines:
24 -> 26

File: docs/guide/content_rendering/embed_and_list_content/list_products_in_catalog.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+The catalog provides a query, based on the filters you configured.
+You can get the query with `$catalog->getQuery()` and use it as criteria in product query (lines 25-27).
```
Comment (juskora) [inline docs/guide/content_rendering/embed_and_list_content/list_products_in_catalog.md:38]: As above,
lines:
25 -> 27
27 -> 29

**Review by juskora — APPROVED** [review]

**Pull request merged**